src/Ting/RepositoryProvider.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CCMBenchmark\Ting\ApiPlatform\Ting;
+
+use CCMBenchmark\Ting\MetadataRepository;
+use CCMBenchmark\Ting\Repository\Metadata;
+use CCMBenchmark\Ting\Repository\Repository;
+use CCMBenchmark\TingBundle\Repository\RepositoryFactory;
+use ReflectionClass;
+
+use function array_key_exists;
+use function class_exists;
+
+/** @internal */
+final class RepositoryProvider
+{
+    /** @var array<class-string, Repository|null> */
+    private array $localCache = [];
+
+    public function __construct(
+        private RepositoryFactory $ting,
+        private MetadataRepository $metadataRepository,
+    ) {
+    }
+
+    /**
+     * @param class-string<T> $resourceClass
+     *
+     * @return Repository<T>|null
+     *
+     * @template T of object
+     */
+    public function getRepositoryFromResource(string $resourceClass): Repository|null
+    {
+        if (array_key_exists($resourceClass, $this->localCache)) {
+            return $this->localCache[$resourceClass];
+        }
+
+        if (! class_exists($resourceClass)) {
+            return $this->localCache[$resourceClass] = null;
+        }
+
+        $repositoryName = null;
+        $retNull = false;
+        $object = (new ReflectionClass($resourceClass))->newInstanceWithoutConstructor();
```

Comment:
Since `MetadataRepository#findMetadataForEntity()` in ting is only doing a `get_class()` on the object, we could modify this method to also accept a `class-string`.